src/fetch.ts

```diff
@@ -0,0 +1,74 @@
+import rawFetch from 'node-fetch';
+
+/**
+ * The type signature for {@link fetch}'s `retryDelay` function.
+ */
+
+export type RequestDelayFunction = (
+  attempt: number,
+  error: Error | null,
+  response: Response | null
+) => number;
+
+/**
+ * The type signature for {@link fetch}'s `retryOn` function.
+ */
+
+export type RequestRetryOnFunction = (
+  attempt: number,
+  error: Error | null,
+  response: Response | null
+) => boolean | Promise<boolean>;
+
+/**
+ * The set of options that can be passed to {@link fetch}, augmented with
+ * a number of non-standard fields related to retrying requests.
+ */
+
+export interface RequestInitWithRetry extends RequestInit {
+  retries?: number;
+  retryDelay?: number | RequestDelayFunction;
+  retryOn?: number[] | RequestRetryOnFunction;
+}
+
+/**
+ * The default exponential backoff delay function used by
+ * {@link fetch} when it needs to retry a request.
+ *
+ * Note that since Functions are limited to 10 seconds of runtime, requests
+ * that use this delay function can only retry a maximum of 4 times (8 seconds).
+ */
+
+export const exponentialDelay = (attempt: number) => Math.pow(2, attempt) * 500;
+
+const fetchBuilder = (): ((
+  input: RequestInfo,
+  init?: RequestInitWithRetry
+) => Promise<Response>) =>
+  require('fetch-retry')(rawFetch, {
```

Comment:
I changed this to an `import`, but this required applying [this fetch-retry PR](https://github.com/jonbern/fetch-retry/pull/65) using `patch-package` to fix the package's types. Let me know if you know of a better way!

Reply:
That's a shame that fetch-retry has that issue in its types. Can't think of a better way to deal with it though.

Reply:
Agreed! And thank you for reaching out to the library author. If they're able to share their test suite, I'll definitely take a look and see what might be causing any problems with `export default`. `runtime-helpers` might be an interesting case study against the type change breaking things in general, since it has to support CommonJS interoperability to work with Functions.